java-checks/src/main/java/org/sonar/java/checks/serialization/SerializableFieldInSerializableClassCheck.java

```diff
@@ -54,14 +60,46 @@ public void visitNode(Tree tree) {
     if (isSerializable(classTree)
       && !SerializableContract.hasSpecialHandlingSerializationMethods(classTree)
       && !classTree.symbol().type().isSubtypeOf("javax.servlet.http.HttpServlet")) {
-      for (Tree member : classTree.members()) {
-        if (member.is(Tree.Kind.VARIABLE)) {
-          checkVariableMember((VariableTree) member);
-        }
-      }
+      
+      Set<String> constructorInjectedParams = getConstructorInjectedFields(classTree);
+      Set<String> setterInjectedParams = getSetterInjectedFields(classTree);
+      
+      classTree.members().stream()
+        .filter(member -> member.is(Tree.Kind.VARIABLE))
+        .map(VariableTree.class::cast)
+        .filter(variableTree -> 
+          !(constructorInjectedParams.contains(variableTree.simpleName().name()) ||
+            setterInjectedParams.contains(variableTree.simpleName().name())))
+        .forEach(this::checkVariableMember);
     }
   }
 
+  private static Set<String> getSetterInjectedFields(ClassTree classTree) {
+    return classTree.members().stream()
+      .filter(member -> member.is(Tree.Kind.METHOD))
+      .map(MethodTree.class::cast)
+      .filter(SerializableFieldInSerializableClassCheck::isAnnotatedWithInject)
+      .map(methodTree -> methodTree.simpleName().name())
+      .filter(name -> name.startsWith("set"))
+      .map(name -> name.substring(3).toLowerCase(Locale.ROOT))
+      .collect(Collectors.toSet());
+  }
+
+  private static Set<String> getConstructorInjectedFields(ClassTree classTree) {
+    return classTree.members().stream()
+      .filter(member -> member.is(Tree.Kind.CONSTRUCTOR))
+      .map(MethodTree.class::cast)
+      .filter(SerializableFieldInSerializableClassCheck::isAnnotatedWithInject)
+      .flatMap(methodTree -> methodTree.parameters().stream())
+      .map(variableTree -> variableTree.simpleName().name())
+      .collect(Collectors.toSet());
+  }
+
+  private static boolean isAnnotatedWithInject(MethodTree methodTree) {
+    return methodTree.symbol().metadata().isAnnotatedWith(JAVAX_INJECT)
+      || methodTree.symbol().metadata().annotations().stream().anyMatch(annotation -> annotation.symbol().isUnknown());
```

Comment:
Without semantic this annotation won't be found. So this line protects of FPs in case of incomplete semantic., the same check is done in line 179